drivers/i2c/i2c_andes_atciic100.c

```diff
@@ -186,14 +189,7 @@ static int i2c_atciic100_transfer(const struct device *dev,
 
 exit:
 	/* Wait for transfer complete */
-	k_sem_take(&dev_data->i2c_busy_sem, K_FOREVER);
-
-	if (dev_data->status.target_ack != 1) {
-		k_sem_give(&dev_data->i2c_busy_sem);
-		return -EIO;
-	}
-	dev_data->status.target_ack = 0;
-	k_sem_give(&dev_data->i2c_busy_sem);
+	k_sem_give(&dev_data->bus_lock);
```

Comment:
This comment is no longer accurate.  The wait for transfer complete is now done by `i2c_atciic100_controller_send/receive()`.